Source/Core/Core/HW/EXI_DeviceIPL.cpp

```diff
@@ -152,15 +152,68 @@ void CEXIIPL::DoState(PointerWrap &p)
 
 void CEXIIPL::LoadFileToIPL(const std::string& filename, u32 offset)
 {
-	File::IOFile pStream(filename, "rb");
-	if (pStream)
+	File::IOFile stream(filename, "rb");
+	if (!stream)
+		return;
+
+	u64 filesize = stream.GetSize();
+
+	stream.ReadBytes(m_pIPL + offset, filesize);
+
+	m_FontsLoaded = true;
+}
+
+std::string CEXIIPL::FindIPLDump(const std::string& path_prefix)
+{
+	std::string ipl_dump_path;
+
+	if (File::Exists(path_prefix + DIR_SEP + USA_DIR + DIR_SEP + GC_IPL))
+		ipl_dump_path = path_prefix + DIR_SEP + USA_DIR + DIR_SEP + GC_IPL;
+	else if (File::Exists(path_prefix + DIR_SEP + EUR_DIR + DIR_SEP + GC_IPL))
+		ipl_dump_path = path_prefix + DIR_SEP + EUR_DIR + DIR_SEP + GC_IPL;
+	else if (File::Exists(path_prefix + DIR_SEP + JAP_DIR + DIR_SEP + GC_IPL))
+		ipl_dump_path = path_prefix + DIR_SEP + JAP_DIR + DIR_SEP + GC_IPL;
+
+	return ipl_dump_path;
+}
+
+void CEXIIPL::LoadFontFile(const std::string& filename, u32 offset)
+{
+	// Official IPL fonts are copyrighted. Dolphin ships with a set of free font alternatives but
+	// unfortunately the bundled fonts have different padding, causing issues with misplaced text
+	// in some titles. This function check if the user has IPL dumps available and load the fonts
+	// from those dumps instead of loading the bundled fonts
+
+	// Check for IPL dumps in User folder
+	std::string ipl_rom_path = FindIPLDump(File::GetUserPath(D_GCUSER_IDX));
+
+	// If not found, check again in Sys folder
+	if (ipl_rom_path.empty())
+		ipl_rom_path = FindIPLDump(File::GetSysDirectory() + GC_SYS_DIR);
+
+	if (File::Exists(ipl_rom_path))
 	{
-		u64 filesize = pStream.GetSize();
+		// The user has an IPL dump, load the font from it
+		File::IOFile stream(ipl_rom_path, "rb");
+		if (!stream)
+			return;
+
+		// Official ANSI and SJIS fonts present on the IPL dumps are 0x2575 and 0x4a24d bytes long,
+		// respectively, so, determine the size of the font being loaded based on the offset
+		u64 fontsize = (offset == 0x1aff00) ? 0x4a24d : 0x2575;
 
-		pStream.ReadBytes(m_pIPL + offset, filesize);
+		INFO_LOG(BOOT, "Found IPL dump, loading %s font from %s", ((offset == 0x1aff00) ? "SJIS" : "ANSI"), (ipl_rom_path).c_str());
```

Comment:
Well, I've chosen ANSI and SJIS to maintain consistency with current code (e.g. the filenames Dolphin currently use for the free fonts and a Panic Alert message further in this file (line 386) uses those terms as well). Do you want me to change this to something else? If so, what terms do you prefer?